h/static/scripts/group-forms/utils/api.ts

```diff
@@ -79,20 +89,30 @@ export class APIError extends Error {
   }
 }
 
-/* Make an API call and return the parsed JSON body or throw APIError. */
+export type APIOptions = {
+  method?: string;
+  json?: object | null;
+  headers?: Record<PropertyKey, unknown>;
+  signal?: AbortSignal;
+
+  /** Index of first item to return in paginated APIs. */
+  offset?: number;
+
+  /** Maximum number of items to return in response for a paginated API. */
+  limit?: number;
```

Comment:
Since this defines the contract for any API call, maybe it makes more sense to define a more generic `queryParams?: Record<PropertyKey, unknown>;` option, rather than `limit` and `offset`.

I get why this is convenient at this point though, so I'll leave it up to you. We can always adjust later.

Reply:
I probably will do this in future. In this PR I added special handling for `limit` and `offset` so callers don't need to know the specific name of the params that are used (`page[offset]` / `page[limit]`).